`QueryBatchOperationTest` is flaky

### <!-- Component -->

<!-- Clients- -->

### <!-- Affected version -->

<!-- -8.8 -->

### <!-- Severity -->

<!-- Low- -->

### <!-- Likelihood -->

<!-- High_ -->

### Description

The `QueryBatchOperationTest` is flaky and fails CI regularly.

### Steps to reproduce

1. Run `QueryBatchOperationTest` repeatedly


### Current behavior

The test occasionally fails with the following exception:

```
io.camunda.client.api.command.ClientException: io.camunda.client.api.command.MalformedResponseException: Failed to parse response body; however, request returned 200: OK
	at io.camunda.client.impl.http.HttpCamundaFuture.unwrapExecutionException(HttpCamundaFuture.java:87)
	at io.camunda.client.impl.http.HttpCamundaFuture.join(HttpCamundaFuture.java:43)
	at io.camunda.client.batchoperation.QueryBatchOperationTest.shouldGetBatchOperationByKey(QueryBatchOperationTest.java:39)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: io.camunda.client.api.command.MalformedResponseException: Failed to parse response body; however, request returned 200: OK
	at io.camunda.client.impl.http.ApiCallback.completeResponse(ApiCallback.java:163)
	at io.camunda.client.impl.http.ApiCallback.handleSuccessResponse(ApiCallback.java:156)
	at io.camunda.client.impl.http.ApiCallback.completed(ApiCallback.java:60)
	at io.camunda.client.impl.http.ApiCallback.completed(ApiCallback.java:32)
	at org.apache.hc.core5.concurrent.BasicFuture.completed(BasicFuture.java:148)
	at org.apache.hc.core5.concurrent.ComplexFuture.completed(ComplexFuture.java:72)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2$1.completed(InternalAbstractHttpAsyncClient.java:321)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer$1.completed(AbstractAsyncResponseConsumer.java:101)
	at org.apache.hc.core5.http.nio.entity.AbstractBinAsyncEntityConsumer.completed(AbstractBinAsyncEntityConsumer.java:87)
	at org.apache.hc.core5.http.nio.entity.AbstractBinDataConsumer.streamEnd(AbstractBinDataConsumer.java:83)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer.streamEnd(AbstractAsyncResponseConsumer.java:142)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.streamEnd(HttpAsyncMainClientExec.java:283)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.dataEnd(ClientHttp1StreamHandler.java:285)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.dataEnd(ClientHttp1StreamDuplexer.java:371)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:340)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:143)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:176)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:125)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:92)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: No enum constant io.camunda.client.api.search.enums.BatchOperationType.UNKNOWN_DEFAULT_OPEN_API
	at java.base/java.lang.Enum.valueOf(Enum.java:293)
	at io.camunda.client.api.search.enums.BatchOperationType.valueOf(BatchOperationType.java:18)
	at io.camunda.client.impl.search.response.BatchOperationImpl.<init>(BatchOperationImpl.java:59)
	at io.camunda.client.impl.search.response.SearchResponseMapper.toBatchOperationGetResponse(SearchResponseMapper.java:179)
	at io.camunda.client.impl.http.ApiCallback.completeResponse(ApiCallback.java:161)
	... 23 more
```

### Expected behavior

The test always passes

### Environment

SM

### Version

- Component Version: 8.8-SNAPSHOT


### Rootcause

The batch operation response in the Java SDK needs to consider receiving unsupported values.

### Solution ideas

Use the `EnumUtil` in the Java SDK to map incoming enum values to the client target type.

### Dev -> QA handover

_No response_

### Links

_No response_

--- a/clients/java/src/main/java/io/camunda/client/impl/search/response/BatchOperationImpl.java
+++ b/clients/java/src/main/java/io/camunda/client/impl/search/response/BatchOperationImpl.java
@@ -19,6 +19,7 @@ import io.camunda.client.api.search.enums.BatchOperationState;
 import io.camunda.client.api.search.enums.BatchOperationType;
 import io.camunda.client.api.search.response.BatchOperation;
 import io.camunda.client.api.search.response.BatchOperationError;
+import io.camunda.client.impl.util.EnumUtil;
 import io.camunda.client.protocol.rest.BatchOperationCreatedResult;
 import io.camunda.client.protocol.rest.BatchOperationResponse;
 import java.util.ArrayList;
@@ -40,10 +41,7 @@ public class BatchOperationImpl implements BatchOperation {
 
   public BatchOperationImpl(final BatchOperationCreatedResult item) {
     batchOperationKey = item.getBatchOperationKey();
-    type =
-        item.getBatchOperationType() != null
-            ? BatchOperationType.valueOf(item.getBatchOperationType().name())
-            : null;
+    type = EnumUtil.convert(item.getBatchOperationType(), BatchOperationType.class);
     status = null;
     startDate = null;
     endDate = null;
@@ -54,10 +52,7 @@ public class BatchOperationImpl implements BatchOperation {
 
   public BatchOperationImpl(final BatchOperationResponse item) {
     batchOperationKey = item.getBatchOperationKey();
-    type =
-        item.getBatchOperationType() != null
-            ? BatchOperationType.valueOf(item.getBatchOperationType().name())
-            : null;
+    type = EnumUtil.convert(item.getBatchOperationType(), BatchOperationType.class);
     status = item.getState() != null ? BatchOperationState.valueOf(item.getState().name()) : null;
     startDate = item.getStartDate();
     endDate = item.getEndDate();

